test/unit/segments_view.cpp

```diff
@@ -131,10 +131,66 @@ struct segments_view_test
         }
     }
 
+    void
+    testNull()
```

Comment:
This is the wrong file you are spreading stuff having to do with queries, paths, fragments, etc.. all in this file which has to do with segments.